src/Services/ResponseService.php

```diff
@@ -2,11 +2,13 @@
 
 namespace Furniture\Services;
 
+use Furniture\Entities\DetailedProductEntity;
+
 class ResponseService
 {
     const UNEXPECTED_ERROR = 'Unexpected error';
 
-    public static function createResponse(string $message, array $data, int $responseCode = 200): array
+    public static function createResponse(string $message, array|DetailedProductEntity $data, int $responseCode = 200): array
```

Comment:
Type hinting to `DetailedProductEntity` breaks the D in SOLID, so change this to `JsonSerializable` instead